src/query/query_executor.c

```diff
@@ -15360,131 +15339,114 @@ qexec_execute_connect_by (THREAD_ENTRY * thread_p, XASL_NODE * xasl, XASL_STATE
 		    {
 		      GOTO_EXIT_ON_ERROR;
 		    }
+		  else if (ev_res != V_TRUE)
+		    {
+		      continue;
+		    }
 		}
-	      else
+
+	      cycle = 0;
+	      /* we found a qualified tuple; now check for cycle */
+	      if (qexec_check_for_cycle (thread_p, xasl->outptr_list, tuple_rec.tpl, &type_list, listfile0, &cycle)
+		  != NO_ERROR)
 		{
-		  ev_res = V_TRUE;
+		  GOTO_EXIT_ON_ERROR;
 		}
 
-	      if (ev_res == V_TRUE)
+	      if (cycle == 0)
+	        {
+		  isleaf_value = 0;
+		}
+
+	      /* found ISLEAF, and we already know LEVEL; we need to add the parent tuple into result list ASAP,
+	       * because we need the information about its position into the list to be kept into each child tuple */
+	      if (!parent_tuple_added)
 		{
-		  if (!connect_by->single_table_opt)
+		  if (listfile1 == connect_by->start_with_list_id)
 		    {
-		      /* fetch rest of regu_var values from input list tuple */
-		      if (fetch_val_list (thread_p, connect_by->regu_list_rest, &xasl_state->vd, NULL, NULL,
-					  input_tuple_rec.tpl, PEEK) != NO_ERROR)
+		      if (!DB_IS_NULL (index_valp) && index_valp->need_clear == true)
 			{
-			  GOTO_EXIT_ON_ERROR;
+			  pr_clear_value (index_valp);
 			}
+
+		      /* set index string pseudocolumn value to tuples from START WITH list */
+		      db_make_string (index_valp, father_index);
 		    }
 
-		  cycle = 0;
-		  /* we found a qualified tuple; now check for cycle */
-		  if (qexec_check_for_cycle (thread_p, xasl->outptr_list, tuple_rec.tpl, &type_list, listfile0, &cycle)
-		      != NO_ERROR)
+		  /* set CONNECT_BY_ISLEAF pseudocolumn value; this is only for completion, we don't know its final
+		   * value yet */
+		  db_make_int (isleaf_valp, isleaf_value);
+
+		  /* preserve the parent position pseudocolumn value */
+		  if (qexec_get_tuple_column_value (tuple_rec.tpl,(xasl->outptr_list->valptr_cnt - PCOL_PARENTPOS_TUPLE_OFFSET),
+						    parent_pos_valp, &tp_Bit_domain) != NO_ERROR)
 		    {
 		      GOTO_EXIT_ON_ERROR;
 		    }
 
-		  if (cycle == 0)
+		  /* make the "final" parent tuple */
+		  tuple_rec = temp_tuple_rec;
+		  if (qdata_copy_valptr_list_to_tuple (thread_p, connect_by->prior_outptr_list, &xasl_state->vd,
+						       &tuple_rec) != NO_ERROR)
 		    {
-		      isleaf_value = 0;
+		      GOTO_EXIT_ON_ERROR;
```

Comment:
is not needed er_set ?

Reply:
yes. er_set() is in qdata_copy_valptr_list_to_tuple().

Reply:
This is legacy code. If there is no requirement for error code in this case, I don't think er_set() is not need to add.